### PR TITLE
Add script for publishing to a registry and S3

### PR DESCRIPTION
**Pull request by samuelkarp**
@euank, @aaithal?

### CODE REVIEW

File: scripts/publish [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	fi
+
+	for tag in ${IMAGE_TAG_VERSION} ${IMAGE_TAG_SHA} ${IMAGE_TAG_LATEST}; do
+		echo "Publishing as ecs-agent-${tag}"
+		dryval aws ${profile} s3 cp ${tarball} s3://${s3_bucket}/ecs-agent-${tag}.tar --acl public-read
+		dryval aws ${profile} s3 cp ${tarball_md5} s3://${s3_bucket}/ecs-agent-${tag}.tar.md5 --acl public-read
+	done
+
+	rm ${tarball}
+	rm ${tarball_md5}
```
Comment (euank) [inline scripts/publish]: Missing quotes (it's possible to have a system tmpdir with spaces and this'll break on it)

File: scripts/publish [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	fi
+	for tag in ${IMAGE_TAG_VERSION} ${IMAGE_TAG_SHA} ${IMAGE_TAG_LATEST}; do
+		echo "Tagging as ${image_name}:${tag}"
+		docker tag -f amazon/amazon-ecs-agent:latest "${image_name}:${tag}"
+		dryval docker push "${image_name}:${tag}"
+	done
+}
+
+save_and_publish_s3() {
+	tarball=$(mktemp)
```
Comment (euank) [inline scripts/publish]: Quotes around $()

**Comment by euank** [comment]
LGTM

**Comment by aaithal** [comment]
:+1:

**Pull request merged**